src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -79,6 +80,8 @@ static ClusterStateMachine()
 
         public event EventHandler<(long Index, string Type)> ValueChanged;
 
+        public ConcurrentDictionary<string,TaskCompletionSource<object>> DeletionCompletedTask = new ConcurrentDictionary<string, TaskCompletionSource<object>>();
```

Comment:
No, we already have a way for us to wait until the cluster is updated, we can just use that.